ably/auth.go

```diff
@@ -266,8 +271,13 @@ func (a *Auth) Authorize(ctx context.Context, params *TokenParams, setOpts ...Au
 		opts = applyAuthOptionsWithDefaults(setOpts...)
 	}
 	a.mtx.Lock()
-	defer a.mtx.Unlock()
-	return a.authorize(ctx, params, opts, true)
+	token, err := a.authorize(ctx, params, opts, true)
```

Comment:
instead of `authorize`, should be renamed to `getAuthToken`, because it certainly doesn't do proper authentication

Reply:
I was going through the file, maybe we need to rename methods that doesn't do actual authorization, some of the code is being used for token related validation. Maybe we should extract it out to be more clear about what's happening behind the scenes

Reply:
What do you mean it doesn't do proper authentication?

In any case, could we move this to a separate pull request? This code was already there.

Reply:
Yes, that would be great, or rather I feel its a small change, we can refactor it if we can